### ISSUE TITLE
spurious SC2086 warning on simple assignment

### ISSUE THREAD

**Issue by viavi-ab** [koalaman/shellcheck#3144]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#! /bin/sh

f() {
        # shellcheck disable=SC3043 # busybox sh supports "local"
        local a=$1
        echo "$a"
}
```

#### Here's what shellcheck currently says:
```
Line 5	SC2086: Double quote to prevent globbing and word splitting.
```
#### Here's what I wanted or expected to see:

Nothing. There's nothing wrong with the assignment, and shellcheck behaves correctly if dialect is set to bash; there may be an interaction with the silencing of SC3043
